home/serializers_v3.py

```diff
@@ -0,0 +1,433 @@
+    elif platform == "sms":
+        messages_length = len(content_page.sms_body._raw_data) - 1
+    elif platform == "ussd":
+        messages_length = len(content_page.ussd_body._raw_data) - 1
+    elif platform == "viber":
+        messages_length = len(content_page.viber_body._raw_data) - 1
+    elif platform == "messenger":
+        messages_length = len(content_page.messenger_body._raw_data) - 1
+    else:
+        return None
+    if messages_length != 0 and message_index > 0:
+        return message_index
+
+
+class RelatedPagesField(serializers.Field):
+    def get_attribute(self, instance):
+        return instance
+
+    def to_representation(self, page):
+        request = self.context["request"]
+        return related_pages_field_representation(page, request)
+
+
+def get_related_page_as_content_page(page):
+    if page.id:
+        return ContentPage.objects.filter(id=page.id).first()
+
+
+def related_pages_field_representation(page, request):
+    related_pages = []
+    for related in page.related_pages:
+        related_page = get_related_page_as_content_page(related.value)
+        title = related_page.title
+        if "whatsapp" in request.GET and related_page.enable_whatsapp is True:
+            if related_page.whatsapp_title:
+                title = related_page.whatsapp_title
+        elif "sms" in request.GET and related_page.enable_sms is True:
+            if related_page.sms_title:
+                title = related_page.sms_title
+        elif "ussd" in request.GET and related_page.enable_ussd is True:
+            if related_page.ussd_title:
+                title = related_page.ussd_title
+        elif "messenger" in request.GET and related_page.enable_messenger is True:
+            if related_page.messenger_title:
+                title = related_page.messenger_title
+        elif "viber" in request.GET and related_page.enable_viber is True:
+            if related_page.viber_title:
+                title = related_page.viber_title
+
+        related_pages.append(
+            {
+                "id": related.id,
+                "value": related_page.id,
+                "title": title,
+            }
+        )
+    return related_pages
+
+
+class BodyField(serializers.Field):
+    """
+    Serializes the "body" field.
+
+    Example:
+    "body": {
+        "message": 1,
+        "next_message": 2,
+        "previous_message": None,
+        "total_messages": 3,
+        "text": "body based on platform requested"
+    }
+    """
+
+    def get_attribute(self, instance):
+        return instance
+
+    def to_representation(self, page):
+        request = self.context["request"]
+        return body_field_representation_V3(page, request)
+
+
+def body_field_representation_V3(page, request):
+    message = 0
+    if "whatsapp" in request.GET and (
+        page.enable_whatsapp is True
+        or ("qa" in request.GET and request.GET["qa"] == "True")
```

Comment:
I think this is another place with the query params where things are wrong that can be solved in the next PR. This checks if whatsapp is enabled or if we've asked for QA. Instead, if it's QA, we should be checking if the latest revision has whatsapp enabled (ideally we would just pass the latest revision to this function, and this function doesn't need to worry about QA)

Reply:
Yes, will tackle in next PR